Compilation errors when using array indexing

Hit another strange issue. I think it doesn't like the array being defined by indexing multiple int positions, then being passed into a function and used for the return value.

Reduced repro:
```
import numpy as np

def bar(x):
    return x[0]

# pythran export foo()
def foo():
    arr = np.array([1, 4, 2, 3, 0])

    order = np.argsort(arr)
    arr = arr[order]

    return bar(arr)
```

Output with `-v`:
```
INFO:      sys file exists: /Users/evan/miniforge3/envs/project/lib/python3.9/site-packages/pythran/pythran.cfg
INFO: platform file exists: /Users/evan/miniforge3/envs/project/lib/python3.9/site-packages/pythran/pythran-darwin.cfg
INFO:     user file exists: /Users/evan/.pythranrc
INFO: pythranrc section [pythran] is valid and options are correct
INFO: pythranrc section [typing] is valid and options are correct
INFO: pythranrc section [backend] is valid and options are correct
INFO: pythranrc section [compiler] is valid and options are correct
running build_ext
DEBUG: new_compiler returns <class 'distutils.unixccompiler.UnixCCompiler'>
building 'a' extension
INFO: C compiler: clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/evan/miniforge3/envs/project/include -arch arm64 -I/Users/evan/miniforge3/envs/project/include -fPIC -O2 -isystem /Users/evan/miniforge3/envs/project/include -arch arm64

creating /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmppy4_djnv/var
creating /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmppy4_djnv/var/folders
creating /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmppy4_djnv/var/folders/l5
creating /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmppy4_djnv/var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq
creating /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmppy4_djnv/var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T
INFO: compile options: '-DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_OPENBLAS -I/opt/homebrew/opt/openblas/include -I/Users/evan/miniforge3/envs/project/lib/python3.9/site-packages/pythran -I/Users/evan/miniforge3/envs/project/lib/python3.9/site-packages/numpy/core/include -I/Users/evan/miniforge3/envs/project/include/python3.9 -c'
extra options: '-std=c++11 -fno-math-errno -Wno-unused-function'
INFO: clang: /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmpj2oxhfah.cpp
/var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmpj2oxhfah.cpp:38:31: error: implicit instantiation of undefined template 'std::tuple_element<0, (anonymous namespace)::pythonic::types::numpy_vexpr<(anonymous namespace)::pythonic::types::ndarray<long long, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 5>>>, (anonymous namespace)::pythonic::types::ndarray<long, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 5>>>>>'
        typedef typename std::tuple_element<0,typename std::remove_reference<__type12>::type>::type __type13;
                              ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__tuple:56:62: note: template is declared here
template <size_t _Ip, class _Tp> struct _LIBCPP_TEMPLATE_VIS tuple_element;
                                                             ^
1 error generated.
WARNING: Compilation error, trying hard to find its origin...
WARNING: Nope, I'm going to flood you with C++ errors!
CRITICAL: Cover me Jack. Jack? Jaaaaack!!!!
E: error: Command "clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/evan/miniforge3/envs/project/include -arch arm64 -I/Users/evan/miniforge3/envs/project/include -fPIC -O2 -isystem /Users/evan/miniforge3/envs/project/include -arch arm64 -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_OPENBLAS -I/opt/homebrew/opt/openblas/include -I/Users/evan/miniforge3/envs/project/lib/python3.9/site-packages/pythran -I/Users/evan/miniforge3/envs/project/lib/python3.9/site-packages/numpy/core/include -I/Users/evan/miniforge3/envs/project/include/python3.9 -c /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmpj2oxhfah.cpp -o /var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmppy4_djnv/var/folders/l5/210k0yvw8xj04j001s6yhlr00000gq/T/tmpj2oxhfah.o -std=c++11 -fno-math-errno -Wno-unused-function" failed with exit status 1

```

Just printing `x` instead of returning something from it makes the problem go away. Using version `0.13.1` on mac.

Thanks in advance!

--- a/pythran/pythonic/include/types/ndarray.hpp
+++ b/pythran/pythonic/include/types/ndarray.hpp
@@ -741,6 +741,12 @@ namespace std
     using type =
         decltype(std::declval<pythonic::types::numpy_gexpr<E, S...>>()[0]);
   };
+
+  template <size_t I, class T, class F>
+  struct tuple_element<I, pythonic::types::numpy_vexpr<T, F>> {
+    using type = decltype(std::declval<pythonic::types::numpy_vexpr<T, F>>()[0]);
+  };
+
 } // namespace std
 
 /* pythran attribute system { */

--- a/pythran/tests/test_ndarray.py
+++ b/pythran/tests/test_ndarray.py
@@ -1066,6 +1066,16 @@ def complex_conversion0(x):
             2 * numpy.arange(10, dtype=numpy.int32).reshape(5,2),
             vexpr0=[List[NDArray[numpy.int32,:,:]], NDArray[numpy.int32,:,:]])
 
+    def test_vexpr1(self):
+        code = '''
+            import numpy as np
+            def vexpr1(a, b):
+                return a[b][0]'''
+        self.run_test(code,
+            numpy.arange(10.),
+            numpy.arange(5, dtype=numpy.int32),
+            vexpr1=[NDArray[float,:], NDArray[numpy.int32,:]])
+
     def test_array_of_pshape(self):
         code = 'def array_of_pshape(x): from numpy import array; return array(x[None].shape)'
         self.run_test(code, numpy.arange(10), array_of_pshape=[NDArray[int,:]])

